File: services/ms-voo/main/java/airportsystem/msvoo/model/Reserva.java
package airportsystem.msvoo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;

@Embeddable
public class Reserva {
    @Column(name = "reserva_id")
    private Long reservaId;
    
    @Column(name = "quantidade")
    private Integer quantidade;
    
    @Column(name = "status")
    private String status;
    
    public Reserva() {}
    
    public Reserva(Long reservaId, Integer quantidade, String status) {
        this.reservaId = reservaId;
        this.quantidade = quantidade;
        this.status = status;
    }
    
    public Long getReservaId() {
        return reservaId;
    }
    
    public void setReservaId(Long reservaId) {
        this.reservaId = reservaId;
    }
    
    public Integer getQuantidade() {
        return quantidade;
    }
    
    public void setQuantidade(Integer quantidade) {
        this.quantidade = quantidade;
    }
    
    public String getStatus() {
        return status;
    }
    
    public void setStatus(String status) {
        this.status = status;
    }
}

File: services/ms-reserva/src/main/kotlin/com/tads/airport_system/ms_reserva/model/Reserva.kt
package com.tads.airport_system.ms_reserva.model

import jakarta.persistence.*

@Entity
@Table(name = "reserva")
data class Reserva(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = 0,

    @Column(nullable = false)
    val flightId: Long,

    @Column(nullable = false)
    val dataReserva: String, // Data da reserva

    @Column(nullable = false)
    val status: String // Estado da reserva
) {
    override fun toString(): String {
        return "Reserva(id=$id, flightId=$flightId, dataReserva=$dataReserva, status='$status')"
    }
}